diff --git a/build/boards/cv180x/cv1800b_milkv_duo_sd/dts_riscv/cv1800b_milkv_duo_sd.dts b/build/boards/cv180x/cv1800b_milkv_duo_sd/dts_riscv/cv1800b_milkv_duo_sd.dts
index 9d840d088..f92c414db 100644
--- a/build/boards/cv180x/cv1800b_milkv_duo_sd/dts_riscv/cv1800b_milkv_duo_sd.dts
+++ b/build/boards/cv180x/cv1800b_milkv_duo_sd/dts_riscv/cv1800b_milkv_duo_sd.dts
@@ -11,8 +11,19 @@
 &spi2 {
 	status = "okay";
 
-	spidev@0 {
+	/delete-node/ spidev@0;
+
+	gc9a01: gc9a01@0 {
+		compatible = "galaxycore,gc9a01";
+		reg = <0>;
 		status = "okay";
+		spi-max-frequency = <8000000>;
+		rotate = <0>;
+		fps = <30>;
+		buswidth = <8>;
+		reset-gpios = <&porte 21 GPIO_ACTIVE_LOW>;
+		dc-gpios = <&porte 25 GPIO_ACTIVE_HIGH>;
+		debug = <0x0>;
 	};
 };
 
diff --git a/build/boards/cv180x/cv1800b_milkv_duo_sd/linux/cvitek_cv1800b_milkv_duo_sd_defconfig b/build/boards/cv180x/cv1800b_milkv_duo_sd/linux/cvitek_cv1800b_milkv_duo_sd_defconfig
index 89bfd2936..8dcb6d451 100644
--- a/build/boards/cv180x/cv1800b_milkv_duo_sd/linux/cvitek_cv1800b_milkv_duo_sd_defconfig
+++ b/build/boards/cv180x/cv1800b_milkv_duo_sd/linux/cvitek_cv1800b_milkv_duo_sd_defconfig
@@ -189,6 +189,7 @@ CONFIG_ION_SYSTEM_HEAP=y
 CONFIG_ION_CARVEOUT_HEAP=y
 CONFIG_ION_CMA_HEAP=y
 CONFIG_FB_TFT=y
+CONFIG_FB_TFT_GC9A01=y
 CONFIG_FB_TFT_SSD1306=m
 CONFIG_FB_TFT_ST7789V=y
 # CONFIG_IOMMU_SUPPORT is not set
diff --git a/build/boards/cv180x/cv1800b_milkv_duo_sd/u-boot/cvi_board_init.c b/build/boards/cv180x/cv1800b_milkv_duo_sd/u-boot/cvi_board_init.c
index 74941cb09..1568eee44 100644
--- a/build/boards/cv180x/cv1800b_milkv_duo_sd/u-boot/cvi_board_init.c
+++ b/build/boards/cv180x/cv1800b_milkv_duo_sd/u-boot/cvi_board_init.c
@@ -33,12 +33,12 @@ int cvi_board_init(void)
 
 	// UART 4
 	PINMUX_CONFIG(SD1_GPIO1, UART4_TX);
-	PINMUX_CONFIG(SD1_GPIO0, UART4_RX);
+	PINMUX_CONFIG(SD1_GPIO0, PWR_GPIO_25); // Duo Pin 5, LCD DC
 
 	// SPI
 	PINMUX_CONFIG(SD1_CLK, SPI2_SCK);
 	PINMUX_CONFIG(SD1_CMD, SPI2_SDO);
-	PINMUX_CONFIG(SD1_D0, SPI2_SDI);
+	PINMUX_CONFIG(SD1_D0, PWR_GPIO_21);    // Duo Pin 11, LCD Reset
 	PINMUX_CONFIG(SD1_D3, SPI2_CS_X);
 
 	return 0;
diff --git a/linux_5.10/drivers/staging/fbtft/Kconfig b/linux_5.10/drivers/staging/fbtft/Kconfig
index fa51de89e..5c1e9bcc7 100644
--- a/linux_5.10/drivers/staging/fbtft/Kconfig
+++ b/linux_5.10/drivers/staging/fbtft/Kconfig
@@ -171,6 +171,12 @@ config FB_TFT_ST7735R
 	help
 	  Generic Framebuffer support for ST7735R
 
+config FB_TFT_GC9A01
+	tristate "FB driver for the GC9A01 LCD Controller"
+	depends on FB_TFT
+	help
+	  Framebuffer support for GC9A01 round LCD controllers.
+
 config FB_TFT_ST7789V
 	tristate "FB driver for the ST7789V LCD Controller"
 	depends on FB_TFT
diff --git a/linux_5.10/drivers/staging/fbtft/Makefile b/linux_5.10/drivers/staging/fbtft/Makefile
index aff25cae8..55cd02199 100644
--- a/linux_5.10/drivers/staging/fbtft/Makefile
+++ b/linux_5.10/drivers/staging/fbtft/Makefile
@@ -31,6 +31,7 @@ obj-$(CONFIG_FB_TFT_SSD1305)     += fb_ssd1325.o
 obj-$(CONFIG_FB_TFT_SSD1331)     += fb_ssd1331.o
 obj-$(CONFIG_FB_TFT_SSD1351)     += fb_ssd1351.o
 obj-$(CONFIG_FB_TFT_ST7735R)     += fb_st7735r.o
+obj-$(CONFIG_FB_TFT_GC9A01)      += fb_gc9a01.o
 obj-$(CONFIG_FB_TFT_ST7789V)     += fb_st7789v.o
 obj-$(CONFIG_FB_TFT_TINYLCD)     += fb_tinylcd.o
 obj-$(CONFIG_FB_TFT_TLS8204)     += fb_tls8204.o
diff --git a/linux_5.10/drivers/staging/fbtft/fb_gc9a01.c b/linux_5.10/drivers/staging/fbtft/fb_gc9a01.c
new file mode 100644
index 000000000..1335926cb
--- /dev/null
+++ b/linux_5.10/drivers/staging/fbtft/fb_gc9a01.c
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * FB driver for the GC9A01 LCD Controller.
+ */
+
+#include <linux/bitops.h>
+#include <linux/delay.h>
+#include <linux/init.h>
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <video/mipi_display.h>
+
+#include "fbtft.h"
+
+#define DRVNAME "fb_gc9a01"
+
+#define MADCTL_BGR BIT(3)
+#define MADCTL_MV BIT(5)
+#define MADCTL_MX BIT(6)
+#define MADCTL_MY BIT(7)
+
+static int init_display(struct fbtft_par *par)
+{
+	par->fbtftops.reset(par);
+	mdelay(120);
+
+	write_reg(par, 0xEF);
+	write_reg(par, 0xEB, 0x14);
+	write_reg(par, 0xFE);
+	write_reg(par, 0xEF);
+	write_reg(par, 0xEB, 0x14);
+	write_reg(par, 0x84, 0x40);
+	write_reg(par, 0x85, 0xFF);
+	write_reg(par, 0x86, 0xFF);
+	write_reg(par, 0x87, 0xFF);
+	write_reg(par, 0x88, 0x0A);
+	write_reg(par, 0x89, 0x21);
+	write_reg(par, 0x8A, 0x00);
+	write_reg(par, 0x8B, 0x80);
+	write_reg(par, 0x8C, 0x01);
+	write_reg(par, 0x8D, 0x01);
+	write_reg(par, 0x8E, 0xFF);
+	write_reg(par, 0x8F, 0xFF);
+	write_reg(par, 0xB6, 0x00, 0x00);
+	write_reg(par, MIPI_DCS_SET_PIXEL_FORMAT, MIPI_DCS_PIXEL_FMT_16BIT);
+	write_reg(par, 0x90, 0x08, 0x08, 0x08, 0x08);
+	write_reg(par, 0xBD, 0x06);
+	write_reg(par, 0xBC, 0x00);
+	write_reg(par, 0xFF, 0x60, 0x01, 0x04);
+	write_reg(par, 0xC3, 0x13);
+	write_reg(par, 0xC4, 0x13);
+	write_reg(par, 0xC9, 0x22);
+	write_reg(par, 0xBE, 0x11);
+	write_reg(par, 0xE1, 0x10, 0x0E);
+	write_reg(par, 0xDF, 0x21, 0x0C, 0x02);
+	write_reg(par, 0xF0, 0x45, 0x09, 0x08, 0x08, 0x26, 0x2A);
+	write_reg(par, 0xF1, 0x43, 0x70, 0x72, 0x36, 0x37, 0x6F);
+	write_reg(par, 0xF2, 0x45, 0x09, 0x08, 0x08, 0x26, 0x2A);
+	write_reg(par, 0xF3, 0x43, 0x70, 0x72, 0x36, 0x37, 0x6F);
+	write_reg(par, 0xED, 0x1B, 0x0B);
+	write_reg(par, 0xAE, 0x77);
+	write_reg(par, 0xCD, 0x63);
+	write_reg(par, 0x70, 0x07, 0x07, 0x04, 0x0E, 0x0F, 0x09, 0x07, 0x08, 0x03);
+	write_reg(par, 0xE8, 0x34);
+	write_reg(par, 0x62, 0x18, 0x0D, 0x71, 0xED, 0x70, 0x70,
+		  0x18, 0x0F, 0x71, 0xEF, 0x70, 0x70);
+	write_reg(par, 0x63, 0x18, 0x11, 0x71, 0xF1, 0x70, 0x70,
+		  0x18, 0x13, 0x71, 0xF3, 0x70, 0x70);
+	write_reg(par, 0x64, 0x28, 0x29, 0xF1, 0x01, 0xF1, 0x00, 0x07);
+	write_reg(par, 0x66, 0x3C, 0x00, 0xCD, 0x67, 0x45, 0x45,
+		  0x10, 0x00, 0x00, 0x00);
+	write_reg(par, 0x67, 0x00, 0x3C, 0x00, 0x00, 0x00, 0x01,
+		  0x54, 0x10, 0x32, 0x98);
+	write_reg(par, 0x74, 0x10, 0x85, 0x80, 0x00, 0x00, 0x4E, 0x00);
+	write_reg(par, 0x98, 0x3E, 0x07);
+	write_reg(par, 0x35);
+	write_reg(par, MIPI_DCS_ENTER_INVERT_MODE);
+	write_reg(par, MIPI_DCS_EXIT_SLEEP_MODE);
+	mdelay(120);
+	write_reg(par, MIPI_DCS_SET_DISPLAY_ON);
+	mdelay(120);
+
+	return 0;
+}
+
+static int set_var(struct fbtft_par *par)
+{
+	u8 madctl_par = 0;
+
+	if (par->bgr)
+		madctl_par |= MADCTL_BGR;
+
+	switch (par->info->var.rotate) {
+	case 0:
+		break;
+	case 90:
+		madctl_par |= MADCTL_MV | MADCTL_MY;
+		break;
+	case 180:
+		madctl_par |= MADCTL_MX | MADCTL_MY;
+		break;
+	case 270:
+		madctl_par |= MADCTL_MV | MADCTL_MX;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	write_reg(par, MIPI_DCS_SET_ADDRESS_MODE, madctl_par);
+	return 0;
+}
+
+static int blank(struct fbtft_par *par, bool on)
+{
+	if (on)
+		write_reg(par, MIPI_DCS_SET_DISPLAY_OFF);
+	else
+		write_reg(par, MIPI_DCS_SET_DISPLAY_ON);
+
+	return 0;
+}
+
+static struct fbtft_display display = {
+	.regwidth = 8,
+	.width = 240,
+	.height = 240,
+	.fps = 30,
+	.fbtftops = {
+		.init_display = init_display,
+		.set_var = set_var,
+		.blank = blank,
+	},
+};
+
+FBTFT_REGISTER_DRIVER(DRVNAME, "galaxycore,gc9a01", &display);
+
+MODULE_ALIAS("spi:gc9a01");
+MODULE_ALIAS("platform:gc9a01");
+MODULE_DESCRIPTION("FB driver for the GC9A01 LCD Controller");
+MODULE_AUTHOR("AIStatusHub");
+MODULE_LICENSE("GPL");
